nautobot/dcim/filters.py

```diff
@@ -159,21 +178,85 @@ class SiteFilterSet(NautobotFilterSet, TenancyFilterSet, StatusModelFilterSetMix
         to_field_name="slug",
         label="Region (slug)",
     )
+    has_circuit_terminations = RelatedMembershipBooleanFilter(
+        field_name="circuit_terminations",
+        label="Has circuit terminations",
+    )
+    has_devices = RelatedMembershipBooleanFilter(
+        field_name="devices",
+        label="Has devices",
+    )
+    powerpanels = django_filters.ModelMultipleChoiceFilter(
```

Comment:
Might it be good to add an explanatory comment here, something like?
```python
# The reverse relation here is misnamed as `powerpanel`, but fixing it would be a breaking API change.
# 2.0 TODO: fix the reverse relation name, at which point this filter can be deleted here and added to Meta.fields.
```